### PR TITLE
feat(pd): integrate `pd-service` into hugegraph

### PR DESCRIPTION
**Pull request by VGalaxies**
subtask of #2265

For detailed module analysis documentation, please refer to [this link](https://hugegraph.feishu.cn/wiki/MpWOwFfjfiYyf8kIZ0TcQC9dn8d).

### CODE REVIEW

**Comment by VGalaxies** [comment]
@imbajin Should we remove all content under `license` and `upgrade` packages?

**Comment by VGalaxies** [comment]
> @imbajin Should we remove all content under `license` and `upgrade` packages? Or mark them when merging with the `Useless` annotation from `pd-store` branch later?

@imbajin marked with `Useless` now

File: hugegraph-pd/hg-pd-service/src/main/java/org/apache/hugegraph/pd/rest/API.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.MessageOrBuilder;
+import com.google.protobuf.util.JsonFormat;
+
+public class API {
+
+    public static final String VERSION = "3.6.3";
```
Comment (imbajin) [inline hugegraph-pd/hg-pd-service/src/main/java/org/apache/hugegraph/pd/rest/API.java:34]:
```suggestion
    public static final String VERSION = "3.6.3";
```

do we need change it?

Comment (VGalaxies) [reply hugegraph-pd/hg-pd-service/src/main/java/org/apache/hugegraph/pd/rest/API.java]: The change may have minimal impact; currently reading directly from the pom.

image

File: hugegraph-pd/hg-pd-service/src/main/java/org/apache/hugegraph/pd/rest/API.java [hunk 2]
```diff
@@ -31,7 +32,8 @@
 
 public class API {
 
-    public static final String VERSION = "3.6.3";
+    // Currently reading directly from the pom.
+    public static final String VERSION = VersionUtil.getPomVersion();
```
Comment (imbajin) [inline hugegraph-pd/hg-pd-service/src/main/java/org/apache/hugegraph/pd/rest/API.java:36]: or may consider using this way:
```xml
<build>
    <plugins>
        <plugin>
            <groupId>org.apache.maven.plugins</groupId>
            <artifactId>maven-jar-plugin</artifactId>
            <configuration>
                <archive>
                    <manifest>
                        <addDefaultImplementationEntries>true</addDefaultImplementationEntries>
                    </manifest>
                </archive>
            </configuration>
        </plugin>
    </plugins>
</build>
```

```java
public static String getPomVersion() {
    return VersionUtil.class.getPackage().getImplementationVersion();
}
```

Update the implement in the common-module

**Review by imbajin — APPROVED** [review]
Quickly move ⌚️

**Review by liuxiaocs7 — APPROVED** [review]

**Pull request merged**